api/application/services/protected_domain_service.py

```diff
@@ -45,7 +40,10 @@ def list_protected_domains(self) -> Set[str]:
         return self._list_protected_permission_domains()
 
     def delete_protected_domain_permission(
-        self, domain: str, user_subjects_list: List[str | None]
+        self,
+        domain: str,
+        user_subjects_list: List[str | None],
+        schema_service: "SchemaService",
```

Comment:
I'm not sure I like the idea of us passing the instance of the SchemaService as a parameter into a function of this class, could we move into the constructor like we do for the adapters?